src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets

```diff
@@ -550,6 +556,42 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <Target
+      Name="_BlazorWasmAOT"
+      DependsOnTargets="_EnsureWasmAOTWorkload;_GatherWasmFilesToAOT;WasmBuildApp"
+      BeforeTargets="_ProcessPublishFilesForBlazor"
+      Condition="'$(RunAOTCompilation)' == 'true'">
+
+    <!--
+      AOT compilation produces a copy of the dotnet.js and dotnet.wasm file that are required to run the app. Replace the regular js and wams files with the AOTed version.
+      But first, mangle the AOTed dotnet.js file to include the runtime version.
+    -->
+
+    <Copy SourceFiles="@(WasmNativeAsset)" DestinationFiles="%(WasmNativeAsset.Directory)\dotnet.$(_DotNetJsVersion).js" Condition="%(FileName) == 'dotnet' AND %(Extension) == '.js'">
+      <Output TaskParameter="CopiedFiles" ItemName="_PublishAotDotnetJsFile" />
+    </Copy>
+
+    <ItemGroup>
+      <ResolvedFileToPublish Remove="$(_BlazorDotNetJSFilePath)" />
+      <ResolvedFileToPublish Remove="@(ResolvedFileToPublish)" Condition="%(FileName) == 'dotnet' AND %(Extension) == '.wasm'" />
+
+      <ResolvedFileToPublish Include="@(_PublishAotDotnetJsFile)" AssetType="native" RelativePath="%(FileName)%(Extension)" />
```

Comment:
What does the "AssetType" determine here?

Reply:
We use it in when producing the blazor.boot.json: https://github.com/dotnet/sdk/blob/master/src/BlazorWasmSdk/Tasks/GenerateBlazorWebAssemblyBootJson.cs#L137. It's trying to immitate the dotnet.js added by default.